internal/html/scrape/schema/json-ld/recipe.go

```diff
@@ -83,14 +84,39 @@ func (rp *RecipeProcessor) parseJSON(data string) (map[string]any, error) {
 
 	addSchemaCtx(recipeNode)
 
-	recipeNode, err := rp.proc.Compact(recipeNode, rp.ctx, rp.opts)
+	recipeNode, err = rp.proc.Compact(recipeNode, rp.ctx, rp.opts)
 	if err != nil {
 		return nil, fmt.Errorf("could not compact Recipe node: %w", err)
 	}
 
 	return recipeNode, nil
 }
 
+func unmarshalJSONObjectOrArray(data string) (map[string]any, error) {
+	var m map[string]any
+	if err := json.Unmarshal([]byte(data), &m); err == nil {
+		return m, nil
+	}
+
+	var nodes []any
+	if err := json.Unmarshal([]byte(data), &nodes); err != nil {
+		var syntaxError *json.SyntaxError
+		if errors.As(err, &syntaxError) {
+			return nil, fmt.Errorf("unmarshal as array failed at byte offset %d", syntaxError.Offset)
```

Comment:
Can we also include syntaxError.msg? This will give insight into what the error was at the given offset.

Reply:
sure :)